presto-spi/src/main/java/com/facebook/presto/spi/session/SessionConfigurationContext.java

```diff
@@ -25,12 +27,14 @@
     private final String user;
     private final Optional<String> source;
     private final Set<String> clientTags;
+    private final ResourceGroupId resourceGroupId;
```

Comment:
Question: Is it worth adding queryType here?

Reply:
Not sure. What's the use case for `queryType`? 

I am also thinking to have `String` instead of `ResourceGroupId` here. Thoughts?

Reply:
1. I can think of a few.. we might want to apply session properties only to some classes of queries?
2. Definitely not - we just realized that putting sources into the resource group names makes them liable to have periods in them. We should definitely not be passing string around because its impossible to reverse engineer the structure out of it.